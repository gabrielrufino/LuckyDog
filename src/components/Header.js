import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {useNavigation} from '@react-navigation/native';

import GoBackButton from './GoBackButton';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flexDirection: 'row',
    paddingHorizontal: 10,
    paddingVertical: 10,
  },
  horizontalSpace: {
    paddingHorizontal: 5,
  },
  title: {
    color: '#C8AB83',
    fontSize: 25,
  },
});

export default function Header({title}) {
  const navigation = useNavigation();

  return (
    <View style={styles.container}>
      <GoBackButton onPress={navigation.goBack} />
      <View style={styles.horizontalSpace} />
      <Text style={styles.title}>{title}</Text>
    </View>
  );
}
