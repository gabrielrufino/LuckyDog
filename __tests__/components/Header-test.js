jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(() => ({
    goBack: jest.fn(),
  })),
}));

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../../src/components/Header';

describe('components/Header.js', () => {
  it('Should renders the header correctly', () => {
    const title = 'Header title';

    const tree = renderer.create(<Header title={title} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
