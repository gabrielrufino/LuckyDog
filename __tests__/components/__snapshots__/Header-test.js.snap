// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header.js Should renders the header correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "padding": 15,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#C8AB83",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 5,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#C8AB83",
        "fontSize": 25,
      }
    }
  >
    Header title
  </Text>
</View>
`;
